stdlib/public/core/UnsafePointer.swift

```diff
@@ -580,8 +590,10 @@ public struct UnsafeMutablePointer<Pointee>: _Pointer {
   /// block. The memory must not be initialized or `Pointee` must be a trivial type.
   @inlinable
   public func deallocate() {
-    Builtin.deallocRaw(_rawValue, (-1)._builtinWordValue,
-                       Builtin.alignof(Pointee.self))
+    // Passing zero alignment to the runtime forces aligned
+    // deallocation. This ensures that the runtime's allocation and
+    // deallocation paths are compatible.
+    Builtin.deallocRaw(_rawValue, (-1)._builtinWordValue, (0)._builtinWordValue)
```

Comment:
Shouldn't the alignment here match the value that was used for the allocation?  That is, `if Builtin.alignof(Pointee.self) < _minAllocationAlignment()` then we should use `Builtin.alignof(Pointee.self)`.

Reply:
@compnerd We don't want to pass any alignment to the Swift runtime here. We just want to force it to use the `aligned_free` path, which doesn't take alignment as a parameter.

The alternative would be to have a `Builtin.allignedDealloc` that doesn't take alignment at all, but that would introduce unnecessary complexity in the compiler.

Reply:
I guess my confusion stems from the fact that it seems that there may be a path that is not `aligned_alloc` above (where the `Builtin.sizeof(Pointee.self)` is used .. or did I mix up something in my head and that will also go down the `aligned_alloc` path?

Reply:
Hopefully I removed all places where non-aligned-alloc could be called for user-allocated memory.

`UnsafePointer.allocate`:

```
    var align = Builtin.alignof(Pointee.self)
    // Force manually allocated memory to use the "aligned" allocation
    // path so that deallocation does not require the original
    // alignment. The runtime guarantees "aligned" allocation for either 
    // alignment > _MinAllocationAlignment or alignment == 0.
    if Int(align) <= _minAllocationAlignment() {
      align = (0)._builtinWordValue
    }
    let rawPtr = Builtin.allocRaw(size._builtinWordValue, align)
```

Can you think of any way I can make that comment more clear?

Reply:
I was very confused for a while about the distinction between the alignment value and the alignment mask, since the Swift Builtins weren't commented at all. I've rewritten the comments to be more precise.